sink/sink

```diff
@@ -339,6 +349,10 @@ class Status(object):
         for reporter in self.reporters:
             try:
                 reporter.push(data)
+            except GithubPushError:
```

Comment:
I don't like this at all.  This exception could now be fired at any time at any point from within execution of the program.  You therefore need to be able to catch it from any context that the program might find itself in from the instant the signal handler has been installed.

Probably that should even include the call to signal.signal() itself, since it's even possible that the exception could be thrown before that call returns.

This is part of why I was trying to push you toward calling sys.exit() from the handler directly...